### PR TITLE
Implementable Mythos

### PR DESCRIPTION
**Pull request by petercombs**

### CODE REVIEW

File: eldritch/mythos.py [hunk 1]
```diff
@@ -146,19 +147,123 @@ def get_interrupt(self, event, owner, state):
...
+    super(Mythos19, self).__init__("Mythos19", "Witch", "Cave", {"moon"}, {"plus"})
+  #TODO: close merchant district streets (in or out)
+
+
+class Mythos20(Headline):
+  def __init__(self):
+    super(Mythos20, self).__init__("Mythos20", "Cave", "Roadhouse", {"moon"}, {"plus"})
+
+  def create_event(self, state):
+    return self.ReturnToCup('streets')
```
Comment (fingoldfish) [inline eldritch/mythos.py:264]: I think you mean events.ReturnToCup.

**Review by fingoldfish — CHANGES_REQUESTED** [review]

File: eldritch/mythos.py [hunk 2]
```diff
@@ -146,19 +147,123 @@ def get_interrupt(self, event, owner, state):
...
+    return seq
+
+
+class Mythos14(Environment):
+  def __init__(self):
+    super(Mythos14, self).__init__("Mythos14", "Unnamable", "Woods", {"square", "diamond"}, {"circle"})
+
+  def get_trigger(self, event, owner, state):
+    char = getattr(event, 'character', None)
+    if isinstance(event, events.EndMovement) and char.place.name == "Northside":
```
[the remark this answers is not in the document]
Comment (petercombs) [reply eldritch/mythos.py:205]: If it's an EndMovement event, though, won't that have a character?

File: eldritch/mythos.py [hunk 3]
```diff
@@ -135,30 +136,150 @@ def __init__(self):
...
+
+
+class Mythos9(Environment):
+  def __init__(self):
+    super(Mythos9, self).__init__(
+        "Mythos9", "Cave", "Roadhouse", {"square", "diamond"}, {"circle"}, "mystic"
+    )
+
+  def get_modifier(self, thing, attribute):
+    if attribute == "luck":
```
Comment (fingoldfish) [inline eldritch/mythos.py:173]: Same comment - only applies to characters in the city.

File: eldritch/mythos.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fingoldfish) [reply eldritch/mythos.py:223]: That's my mistake. I didn't read carefully enough. Though I'm curious why this isn't just
if isinstance(...) and event.character.place.name ...

File: eldritch/mythos.py [hunk 4]
```diff
@@ -146,19 +147,123 @@ def get_interrupt(self, event, owner, state):
...
+  def __init__(self):
+    super(Mythos14, self).__init__("Mythos14", "Unnamable", "Woods", {"square", "diamond"}, {"circle"})
+
+  def get_trigger(self, event, owner, state):
+    char = getattr(event, 'character', None)
+    if isinstance(event, events.EndMovement) and char.place.name == "Northside":
+      gain = events.Gain(char, {'clues': 1})
+      check = events.Check(char, 'will', -1)
+      wonders = events.PassFail(char, check, events.Nothing(), events.Loss(char, {'sanity': 1}))
+      return events.Sequence([gain, wonders])
```
[the remark this answers is not in the document]
Comment (fingoldfish) [reply eldritch/mythos.py:227]: Ah, but the character is actually used in a different place - it's used when a character goes insane. You see, when a character goes insane, the event stack has to be cleared of anything the character was doing at the time (e.g. fighting monsters, having an encounter). But you wouldn't want to clear any events caused by something unrelated to the character (e.g. drawing and resolving a mythos card). Sequence events with the character attached to them will get cleared from the stack if the character goes insane, other Sequence events will not.

In this case, I think you do want to add the character to the Sequence - if there are multiple things that can happen when the character ends their movement in the streets, and this particular one causes the character to go insane, then the other events shouldn't happen. I'm not sure what those other events would be, but better to have it and not need it than to need it and not have it.

File: eldritch/mythos.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (petercombs) [reply eldritch/mythos.py:223]: Mainly so the stuff inside the if block can also use the `char` variable, rather than the longer `event.character`

**Pull request merged**